### PR TITLE
Make "int" token values be long type when [Ll] specified.

### PR DESCRIPTION
**Pull request by trotterdylan**

### CODE REVIEW

**Comment by whitequark** [comment]
Uhm, no, I don't think so. Like I said elsewhere, the Python 2 support is provided for completeness, and in general the Python 3 AST format is used. You can notice this in the node structure as well. I see no good reason to introduce additional complexity to the codebase for this.

**Comment by trotterdylan** [comment]
FWIW, this is the last change that was required to get the Grumpy tests passing but I don't mind making this change locally in my repo.

**Comment by whitequark** [comment]
As in you're using a fork of pythonparser? Or in grumpy?

**Comment by trotterdylan** [comment]
There will be a fork in grumpy/third_party/pythonparser regardless. There are a couple minor things I'd like independence on (e.g. no regex library).

Re: differing AST format: that strikes me as a different kind of issue in that all the information about the Python 2 syntax is still represented there, just in a different form. The trouble with the the [Ll] specifier is that there's no way for a Python 2 parser to know whether the specifier was present (I guess you could look at the range information and pull out the trailing character but it's a bit of a mess). The semantics of a parsed program may be different depending on whether [Ll] was different.

**Comment by whitequark** [comment]
> The trouble with the the [Ll] specifier is that there's no way for a Python 2 parser to know whether the specifier was present (I guess you could look at the range information and pull out the trailing character but it's a bit of a mess).

Oh, I haven't realized that. Sure.

**Comment by whitequark** [comment]
I would prefer if you added a new token type for "long" rather than doing this type trickery.

**Comment by trotterdylan** [comment]
Updated based on a new "long" token. When I first read your suggestion I thought it was a good idea. Now that I see the implementation I'm not so sure, but I'll leave it up to you.

One thing that occurred to me is that if you're parsing Python 2 syntax in a v3 interpreter, there is going to be no way to differentiate between long and int since there's only one integer type. That may add support for your initial reaction that this isn't worthwhile. For my purposes I'm parsing Python 2 in v2 so this is an improvement, but this patch doesn't fix the problem in the general case.

File: pythonparser/parser.py [hunk 1]
```diff
@@ -1516,6 +1517,13 @@ def atom_2(self, tok):
     def atom_3(self, tok):
         return ast.Num(n=tok.value, loc=tok.loc)
 
+    @action(Tok("long"))
+    def atom_3_long(self, tok):
+        n = tok.value
+        if sys.version_info < (3,):
+            n = long(tok.value)
```
Comment (whitequark) [inline pythonparser/parser.py]: Why can't the `long` token carry a `long(...)` value?

**Comment by whitequark** [comment]
@trotterdylan Do you think you can move the type cast from the parser into the lexer? I'll merge after that.

**Comment by trotterdylan** [comment]
Sorry for the delay. I've moved the type cast logic into the lexer. PTAL

**Pull request merged**